### PR TITLE
Caseflow | Use EFolderService for all apps

### PR DESCRIPTION
**Pull request by dannysellers**
Following [user feedback](https://dsva.slack.com/archives/C2ZBKPMND/p1518722771000350) mentioning incorrect / inconsistent document counts in Queue and Reader in production, Mark discovered that Queue was fetching the document count from Reader: [`Appeal.number_of_documents_url`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/appeal.rb#L129) was returning the incorrect URL. After a change to [`RequestStore.store[:application]`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/controllers/queue_controller.rb#L5) in Queue, [`Appeal.document_service`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/appeal.rb#L593) no longer returned the correct service (`ExternalApi::EfolderService`) when using Queue.

At this point, there is [no reason](https://dsva.slack.com/archives/C2ZBKPMND/p1518724915000674?thread_ts=1518722771.000350&cid=C2ZBKPMND) for any app not to have access to `ExternalApi::EfolderService`. This removes the `RequestStore.store[:application]` restrictions.

### CODE REVIEW

File: app/models/appeal.rb [hunk 1]
```diff
@@ -592,8 +592,7 @@ def fetched_documents
 
   def document_service
     @document_service ||=
-      if (RequestStore.store[:application] == "reader" || RequestStore.store[:application] == "hearings") &&
-         FeatureToggle.enabled?(:efolder_docs_api, user: RequestStore.store[:current_user])
+      if FeatureToggle.enabled?(:efolder_docs_api, user: RequestStore.store[:current_user])
```
Comment (mdbenjam) [inline app/models/appeal.rb]: We can remove this. It's turned on for everyone now. We can just return EFolderService.

**Comment by amprokop** [comment]
@dannysellers @mdbenjam — what about uploading documents? Certification still needs to do that.

**Comment by dannysellers** [comment]
@mdbenjam So the `fetch_document_file` in `EfolderService` should be lifted from that commit? Or should it just return the `VBMSService.fetch_document_file`?

Also, it looks like [`EfolderService.fetch_document_file`](https://github.com/department-of-veterans-affairs/caseflow/blob/f5edc9edc025b1e04aa86467ca0cc3d9bcea8ffd/app/services/external_api/efolder_service.rb#L5) has a different signature than  [`VBMSService`'s](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/services/external_api/vbms_service.rb#L21)  (because of its call to [`EfolderService.send_efolder_request`](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/services/external_api/efolder_service.rb#L93)) 😕 

Maybe `Appeal.document_service` should just be something like:
```ruby
def document_service
  @document_service ||=
    if ["dispatch", :certification].include? RequestStore.store[:application]
      VBMSService
    else
      EFolderService
    end
end
```

**Comment by mdbenjam** [comment]
Uploading is a good question, I think uploading is fine to keep in Certification. We do all the S3 caching and access checking in eFolder, so I'd like to move the fetch document content calls over there. But, it doesn't have to be in this PR if it's going to be more complicated, which it seems like it will be. Maybe for now, what you have exempting dispatch and certification can work.

File: app/models/appeal.rb [hunk 2]
```diff
@@ -127,7 +127,7 @@ def number_of_documents
   end
 
   def number_of_documents_url
-    if document_service == ExternalApi::EfolderService
+    if Rails.env.production? || Rails.env.staging?
```
Comment (mdbenjam) [inline app/models/appeal.rb]: One of my goals is to make it easy to run caseflow connected to eFolder, then we won't need this. But makes sense for now.

File: app/models/appeal.rb [hunk 3]
```diff
@@ -591,13 +591,7 @@ def fetched_documents
...
 
   def document_service
-    @document_service ||=
-      if (RequestStore.store[:application] == "reader" || RequestStore.store[:application] == "hearings") &&
-         FeatureToggle.enabled?(:efolder_docs_api, user: RequestStore.store[:current_user])
-        EFolderService
-      else
-        VBMSService
-      end
+    @document_service ||= VBMSService
```
Comment (mdbenjam) [inline app/models/appeal.rb]: I think we can go ahead and just remove this method. Doesn't seem to be used anywhere else.

**Pull request merged**